spinn_front_end_common/interface/interface_functions/local_tdma_builder.py

```diff
@@ -138,9 +138,23 @@ def __call__(
                     int(math.ceil(machine_time_step * time_scale_factor)))
 
                 # test timings
-                self._test_timings(
+                reduction = self._test_timings(
                     n_phases, time_between_phases, machine_time_step,
                     time_scale_factor, fraction_of_sending, app_vertex.label)
+                if reduction is not None:
+                    max_reduction = max(max_reduction, reduction)
+        if max_reduction != 0:
+            total_time_available = int(math.ceil(
+                machine_time_step * time_scale_factor * fraction_of_sending))
+            total_time_available += int(math.ceil(
+                machine_time_step * time_scale_factor * fraction_of_waiting))
+            new_time_scale_factor = int(math.floor(
+                (total_time_available * max_reduction) / machine_time_step))
+            if new_time_scale_factor != time_scale_factor:
+                logger.info(
+                    "could reduce fraction of time for sending to {},"
+                    "or time scale factor to {}".format(
+                        max_reduction, new_time_scale_factor))
```

Comment:
Logs should start with a capital.  I guess this might be better to refer to the configuration file to let the user really know what they should do.

Reply:
fixed